### PR TITLE
#2098 change averaging for SPMe+SR

### LINKED ISSUE #2098: SPM(e) with SEI growth or lithium plating
At the moment we average the SEI growth, lithium plating and porosity submodels for Single Particle Models. They don't need to be averaged (preprint available in arXiv very soon), so we should change the default. 

This probably applies to crack and LAM submodels too, but I have not proved it so I will leave them as they are.

### PR DESCRIPTION
**Pull request by brosaplanella**
# Description

Changed the averaging for SEI, lithium plating and porosity submodels for SPM and SPMe. Wait to merge until the citation is updated.

Fixes #2098 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/CITATIONS.txt [hunk 1]
```diff
@@ -299,13 +312,13 @@
 @article{ORegan2021,
   author = {O'Regan, Kieran and Brosa Planella, Ferran and Widanage, W. Dhammika and Kendrick, Emma},
   title = {{Thermal-electrochemical parametrisation of a lithium-ion battery: mapping Li concentration and temperature dependencies}},
-  journal = {Journal of The Electrochemical Society},
+  journal = {Submitted for publication},
```
Comment (brosaplanella) [inline pybamm/CITATIONS.txt:315]: That's not going to JES, so updating it.

**Comment by tinosulzer** [comment]
I have a few questions about this, let's go over it in the Monday meeting

**Comment by brosaplanella** [comment]
> I have a few questions about this, let's go over it in the Monday meeting

I will miss the next two meetings, but there's no rush on this so happy to chat about when I am back.

File: pybamm/models/full_battery_models/lithium_ion/base_lithium_ion_model.py [hunk 2]
```diff
@@ -213,7 +211,7 @@ def set_lithium_plating_submodel(self):
             )
         else:
             self.submodels["lithium plating"] = pybamm.lithium_plating.Plating(
-                self.param, self.x_average, self.options
+                self.param, False, self.options
```
Comment (tinosulzer) [inline pybamm/models/full_battery_models/lithium_ion/base_lithium_ion_model.py:214]: can you make this an option?

**Review by tinosulzer — CHANGES_REQUESTED** [review]
Looks good, just the change discussed in meeting

**Comment by brosaplanella** [comment]
Note that I have updated `ORegan2021` to `ORegan2022` now that it has been published, hence all the files modified.

File: pybamm/models/full_battery_models/lithium_ion/spme.py [hunk 3]
```diff
@@ -41,9 +41,11 @@ class SPMe(SPM):
...
         self, options=None, name="Single Particle Model with electrolyte", build=True
     ):
-        # For degradation models we use the "x-average" form since this is a
-        # reduced-order model with uniform current density in the electrodes
-        self.x_average = True
+        # # Set self.x_average based on "x-average side reactions"
+        # if options.get("x-average side reactions") == "true":
+        #     self.x_average = True
+        # else:
+        #     self.x_average = False
```
Comment (brosaplanella) [inline pybamm/models/full_battery_models/lithium_ion/spme.py:48]: I don't think this is really needed as it is overwritten when initialising SPM anyway

File: pybamm/models/full_battery_models/base_battery_model.py [hunk 4]
```diff
@@ -166,6 +169,10 @@ class BatteryModelOptions(pybamm.FuzzyDict):
                 Which electrode(s) intercalates and which is counter. If "both"
                 (default), the model is a standard battery. Otherwise can be "negative"
                 or "positive" to indicate a half-cell model.
+            * "x-average side reactions": str
+                Whether to average the side reactions (SEI growth, lithium plating and
+                the respective porosity change) over the x-axis in Single Particle
+                Models, can be "false" (default) or "true".
```
Comment (tinosulzer) [inline pybamm/models/full_battery_models/base_battery_model.py:175]: shouldn't it be true for SPM? I think it won't make a difference to results, but will be faster

**Review by tinosulzer — APPROVED** [review]
Thanks, looks good!

File: pybamm/models/full_battery_models/base_battery_model.py (hunk 4 above)
Comment (brosaplanella) [reply pybamm/models/full_battery_models/base_battery_model.py]: I think I have fixed it now, take a look. It is a bit messy at the moment, hopefully will be clearer once we have derived the SPMe for cracks and loss of active material submodels.

**Review by tinosulzer — APPROVED** [review]
merge whenever ready

**Pull request merged**